chain/sync.go

```diff
@@ -542,9 +542,11 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 
 	params := s.wallet.ChainParams()
 
+	ntfnCtx, ntfnCtxCancel := context.WithCancel(context.Background())
+	defer ntfnCtxCancel()
```

Comment:
Shouldn't `ntfnCtx` be derived from `ctx` (not `context.Background()`) ?

Reply:
this is from a pr it is rebased over, please comment on #2463 (it's already been reviewed though, i'm just waiting on the dcrd changes to land first).